File: lib/Environment/DevEnvironmentBase/src/main/java/dev/environment/annotation/Environment.java
package dev.environment.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * detail: 环境配置 ( 注解标记类 )
 * @author Ttt
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.CLASS)
public @interface Environment {

    /**
     * 当前 {@link Environment} 具体配置值
     * @return {@link Environment} 具体配置值
     */
    String value();

    /**
     * 当前 {@link Environment} 别名
     * @return {@link Environment} 别名
     */
    String alias() default "";

    /**
     * 是否 Release 环境
     * <pre>
     *     同一个 {@link Module} 只能存在一个 Release 环境配置
     *     用于构建 Release 常量实体类
     * </pre>
     * @return {@code true} yes, {@code false} no
     */
    boolean isRelease() default false;
}

File: application/DevHttpManagerUse/src/main/java/afkt/httpmanager/use/network/config/environment.kt
package afkt.httpmanager.use.network.config

import dev.environment.annotation.Environment
import dev.environment.annotation.Module

// ==============
// = 环境配置信息 =
// ==============

/**
 * detail: Environment 配置
 * @author Ttt
 */
private class EnvironmentConfig private constructor() {

    // ==========
    // = Module =
    // ==========

    @Module(alias = "媒体信息模块")
    private inner class Media {
        @Environment(
            value = "https://raw.githubusercontent.com/afkT/DevUtils/refs/heads/master/application/DevRetrofitUse/src/main/",
            isRelease = true,
            alias = "线上环境"
        )
        private val release: String? = null

        @Environment(value = "https://debug.github.com/", alias = "测试环境")
        private val debug: String? = null

        @Environment(value = "https://development.github.com/", alias = "开发环境")
        private val development: String? = null
    }
}